google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -549,13 +556,37 @@ public Integer getDaysSinceNoncurrentTime() {
 
       /**
        * Returns the date in RFC 3339 format with only the date part (for instance, "2013-01-15").
+       *
+       * @deprecated {@link #getNoncurrentTimeBeforeOffsetDateTime()}
        */
+      @Deprecated
       public DateTime getNoncurrentTimeBefore() {
+        if (noncurrentTimeBefore == null) return null;
+        return new DateTime(millisOffsetDateTimeCodec.decode(noncurrentTimeBefore));
+      }
+
+      /**
+       * Returns the date in RFC 3339 format with only the date part (for instance, "2013-01-15").
+       */
+      public OffsetDateTime getNoncurrentTimeBeforeOffsetDateTime() {
         return noncurrentTimeBefore;
       }
 
-      /* Returns the date in RFC 3339 format with only the date part (for instance, "2013-01-15").*/
+      /**
+       * Returns the date in RFC 3339 format with only the date part (for instance, "2013-01-15").
+       *
+       * @deprecated Use {@link #getCustomTimeBeforeOffsetDateTime()}
+       */
+      @Deprecated
       public DateTime getCustomTimeBefore() {
+        if (customTimeBefore == null) return null;
+        return new DateTime(millisOffsetDateTimeCodec.decode(customTimeBefore));
```

Comment:
I think these two lines can be replaced with:
```suggestion
        return Utils.dateTimeCodec.nullable().encode(customTimeBefore);
```